### PR TITLE
add e2e test for knative

### PR DESCRIPTION
**Pull request by ilackarms**
Signed-off-by: Scott Weiss <sdw35@cornell.edu>

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@
 # TODO: avoid using revision here and use >= semver when go-utils has a release.
 [[constraint]]
   name = "github.com/solo-io/go-utils"
-  revision = "67428b92743fd710ab708665d5fff86e06ff1612"
```
Comment (rickducott) [inline Gopkg.toml]: Should tag a release and depend on some version

Comment (ilackarms) [reply Gopkg.toml]: perhaps we can remove the constraint entirely and simply allow `dep ensure -update` to update the lockfile when needed?

File: test/helpers/build_test_images.go [hunk 2]
```diff
@@ -4,16 +4,11 @@ import (
...
 	"github.com/solo-io/solo-kit/test/helpers"
 )
 
-var glooComponents = []string{
-	"gloo",
-	"discovery",
-	"gateway",
-	"ingress",
-}
 var versionTag = ""
```
[the remark this answers is not in the document]
Comment (ilackarms) [reply test/helpers/build_test_images.go]: because it is mutable

[the remark this answers is not in the document]
Comment (rickducott) [reply test/helpers/build_test_images.go]: oops i missed the reference 👍

File: Gopkg.toml (hunk 1 above)
Comment (rickducott) [reply Gopkg.toml]: I'd be cool with that too 👍

**Pull request merged**